### ISSUE TITLE
Fleet management: Assign rovers to specific buildings

### ISSUE THREAD

**Issue by martin-ez** [martin-ez/project-titan#7]
Nothing in this game has ever put a rover on a road. #70 gave a rover a segment and a distance along it, #5 drove it down a route it was handed, #6 let it find one for itself — and every one of them spawns its rover in a test. #24 closes with a mechanism nobody can watch run. This is where a rover becomes something the player has.

The lever is how many rovers serve an input, never which road they take. #134 gives a building a port at a named corner of its tile for each thing it takes and each thing it makes; each input port is given a rover count and a source port, and every rover assigned to it shuttles: drive to the source, take on a load, drive back, hand it over, and do it again. Which road that is stays #6's answer. A player who routes a fleet by hand routes around the problem the game is about, and invariant 1 is why there is a problem at all — everything a building receives arrives on a rover that drove a road to get there, so the only way to move more is to put more rovers on the road the player built and live with what that does to it.

The source is named rather than found, and only for now. A rover looking for "the port that makes what this input takes" needs inputs, products and recipes, and those are #63 and #10 under #26 — an epic that waits on this one. #132 is where the naming is replaced by the finding, once there is something to find. Until then a load is #70's opaque `Cargo` and the source is a port the player pointed at, which is enough for a fleet to form, to queue, and for #8 to jam it.

Spawning is instant and retiring is not. A rover added to an input's count is on the road on the tick the player asks for it, because the count is the building's own record rather than a fact about a frame, which is what invariant 2 leaves a player edit free to write. A rover taken away finishes what it is doing and leaves the world the next time it reaches the port it serves. The alternative is a load that stops existing halfway down a road, and goods that vanish rather than arrive are the free transfer invariant 1 exists to forbid.

Where the player makes that edit is #59's, so this waits on it. There is no widget layer anywhere in `src/` — `legend.rs` is debug text standing on the diagnostics overlay and says outright that choosing the layer a player reads belongs to #59. A count reached only through a debug keybinding would satisfy invariant 5 and would still leave the first loop in this game one nobody can actually play. #59 has no blockers of its own and can be built alongside #134, so the gate costs ordering and nothing else.

An input whose source no road reaches keeps its rovers parked at home rather than sending them out to strand. #6 answers an unroutable destination with `Stranded`, which is the right answer for one delivery and the wrong one for a standing assignment: a fleet that strands itself the first time a road is bulldozed never recovers when the road comes back.

The fleet-scale corollary lands on the assignment rather than on the driving. An input holds a count and a handful of rovers; what must not appear is a system walking every rover on the map every tick to work out which port each one belongs to.

### Done when

An input port given a count and a source has that many rovers on the road serving it; each drives to the source port, takes on a load, drives back, hands it over and sets off again; two inputs on one building are served by two fleets that do not draw on each other; raising a count puts more rovers on the road on the tick it is raised; lowering it takes them off as they get back rather than where they stand, and nothing they were carrying is lost; an input whose source no road reaches keeps its rovers parked and sets them going again once one does; and the same chain run for a fixed number of ticks delivers measurably more with two rovers assigned to it than with one.

**Comment by martin-ez** [comment]
Missed on the first pass of #40. "Rovers assigned to a building" is a one-to-many relationship, which is what `#[relationship]` and `#[relationship_target]` have modelled natively since 0.16: an `AssignedTo(Entity)` on the rover and its collection on the building, kept in sync in both directions by the engine and cleaned up when either side despawns.

The despawn half is the part worth having. A fleet assignment held as a plain `Entity` field leaves a dangling reference the moment a building is demolished, and that is a bug that shows up as a rover driving to a place that is not there any more rather than as a panic. The relationship machinery makes it not representable instead of making it caught.

The same note is on #4 for road segments, including the correction that 0.19 self-referential relationships are a different feature from the one that models a chain.

**Comment by martin-ez** [comment]
This had no body. Proposing a done-when, and the same correction as #5: the title says "buildings" and means endpoints. Until #26, an endpoint is a box with a cargo token and no recipe.

This is the issue that turns one rover on a road into a fleet, which is what makes every other traffic issue interesting — one rover cannot jam anything.

What it owns:

- A rover belongs to a route or a pair of endpoints rather than wandering. The player decides which, and that decision is the only thing here the player actually makes.
- How many. Adding a rover to an over-served route is how a player discovers the road is the constraint rather than the fleet, and that is the lesson the whole game is built to teach.
- What an idle rover does. Parking is written off in #4's notes, so an unassigned rover either waits at an endpoint occupying nothing, or occupies road and becomes part of the problem. That is a real design choice and it belongs here.
- Fleet scale starts here in earnest: this is the issue that puts thousands of rovers on the map, so a per-tick system that scans every rover to find the few it wants stops being acceptable at exactly this point.

### Done when

Rovers assigned to a route drive it without further instruction, adding one increases what the route delivers over a fixed number of ticks until the road rather than the fleet is the limit, and that ceiling is a number a test asserts.

**Comment by martin-ez** [comment]
This is where the player sets a rover count, and there was nowhere for them to set it. The body says when the edit lands — invariant 2's carve-out, the count being the building's own record rather than a fact about a frame — and nothing about how it is made. There is no widget layer anywhere in `src/`: `legend.rs` is debug text on the diagnostics footing and says outright that choosing the layer a player reads is #59's.

So this now needs #59. The alternative considered was a keybinding plus the debug gizmo view on a selected building, which invariant 5 would allow — whatever it takes to see a system run and interact with it while it runs is part of building it. It was not taken because #59 has no blockers and can be built alongside #134, so gating this on it costs no wall-clock, and because a rover count reached through a debug binding is a first playable loop the player cannot actually play.

This is also why #58 no longer waits on #25: its own body always said it comes off #23 because a panel does not need roads, while the edge said otherwise. With #7 now needing #59, the old edge would have been a cycle.

**Comment by martin-ez** [comment]
Claimed by `feat/7-fleet-management-assign-rovers-to-specif` via `scripts/track.sh claim`.

**Comment by martin-ez** [comment]
Built by `feat/7-fleet-management-assign-rovers-to-specif`, waiting on a human merge of #148.
